snapcraft/internal/project_loader/_config.py

```diff
@@ -219,6 +220,14 @@ def __init__(self, project: project.Project) -> None:
             properties=self.data, project=project
         )
 
+        if project._snap_meta.package_repositories and not os.getenv(
```

Comment:
is this a safeguard? sys.exit(1) is not friendly for testing or figuring things out. If this is just a guard a RuntimeError could be better

Reply:
If not, project sanity checks
and lastly, a new exception

Reply:
Unfortunately `_is_legacy` will catch the exception here.

Reply:
Moved into conduct_project_sanity_check, that works.